geo/src/algorithm/concavehull.rs

```diff
@@ -0,0 +1,418 @@
+use crate::algorithm::convexhull::ConvexHull;
+use crate::algorithm::euclidean_distance::EuclideanDistance;
+use crate::algorithm::euclidean_length::EuclideanLength;
+use crate::prelude::Centroid;
+use crate::utils::partial_min;
+use crate::{Line, LineString, MultiLineString, MultiPoint, MultiPolygon, Point, Polygon};
+use num_traits::Float;
+use rstar::{RTree, RTreeNum};
+use std::collections::VecDeque;
+use std::ops::Add;
+
+pub trait ConcaveHull<T> {
+    /// Returns the concave hull of a Polygon.
```

Comment:
Thank you for adding some doc comments. 

Since you've implemented this for more than just Polygons, maybe clearer to say something like:

> Returns the concave hull of a **Geometry**.

Also bonus points if you can avoid using the term "concave hull" in your description. Maybe something like...

> Returns a polygon which covers the geometry. Unlike _convex_ hulls, which also cover their geometry, a _concave_ hull does so while trying to minimize its area by "shrink wrapping" around the geometry it covers.

Heh - do people know what "shrink wrapping" means? Anyway, good docs are tough, and somewhat subjective, so take it or leave it.

Reply:
What do you think about this?
'''
///Returns a polygon which covers a geometry. Unlike convex hulls, which also cover
/// their geometry, a concave hull does so while trying to further minimize its area by
/// constructing edges such that the exterior of the polygon incorporates points that would
/// be interior points in a convex hull.
'''

Reply:
👏 👏 👏